docs/topics/typecasts.md

```diff
@@ -110,16 +254,16 @@ Smart casts can be used in the following conditions:
 
 ## "Unsafe" cast operator
 
-Usually, the cast operator throws an exception if the cast isn't possible. Thus, it's called _unsafe_.
-The unsafe cast in Kotlin is done by the infix operator `as`.
+To explicitly cast an object to a non-nullable type, use the *unsafe* cast operator `as`:
 
 ```kotlin
 val x: String = y as String
 ```
 
-Note that `null` cannot be cast to `String`, as this type is not [nullable](null-safety.md).
-If `y` is null, the code above throws an exception.
-To make code like this correct for null values, use the nullable type on the right-hand side of the cast:
+If the cast isn't possible, the compiler throws an exception. This is why it's called _unsafe_.
+
+In the previous example, if `y` is `null`, the code above also throws an exception. This is because `null` can't be cast to `String`, as `null` isn't [nullable](null-safety.md).
```

Comment:
I think it's supposed to say `String` isn't nullable.
```suggestion
In the previous example, if `y` is `null`, the code above also throws an exception. This is because `null` can't be cast to `String`, as `String` isn't [nullable](null-safety.md).
```

Reply:
Thanks @lukellmann ! You're right. I fixed this as part of #4247